docs/tutorials/custom_segmentation_trainer.ipynb

```diff
@@ -0,0 +1,524 @@
+    "## Train model\n",
+    "\n",
+    "The remainder of the turial is straightforward and follows the typical [PyTorch Lightning](https://lightning.ai/) training routine. We instantiate a `DataModule` for the LandCover.AI 100 dataset (a small version of the LandCover.AI dataset for notebook testing), instantiate a `CustomSemanticSegmentationTask` with a U-Net and ResNet-50 backbone, then train the model using a Lightning trainer."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dm = LandCoverAI100DataModule(root='data', batch_size=10, num_workers=2, download=True)\n",
+    "\n",
+    "# You can use the following for actual training runs\n",
+    "# from torchgeo.datamodules import LandCoverAIDataModule\n",
+    "# dm = LandCoverAIDataModule(root='data', batch_size=64, num_workers=8, download=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Downloading: \"https://download.pytorch.org/models/resnet50-19c8e357.pth\" to /home/davrob/.cache/torch/hub/checkpoints/resnet50-19c8e357.pth\n",
+      "100%|██████████| 97.8M/97.8M [00:01<00:00, 85.1MB/s]\n"
+     ]
+    }
+   ],
+   "source": [
+    "task = CustomSemanticSegmentationTask(\n",
+    "    model='unet',\n",
+    "    backbone='resnet50',\n",
+    "    weights=True,\n",
+    "    in_channels=3,\n",
+    "    num_classes=6,\n",
+    "    loss='ce',\n",
+    "    lr=1e-3,\n",
+    "    tmax=50,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "\"backbone\":        resnet50\n",
+       "\"class_weights\":   None\n",
+       "\"eta_min\":         1e-06\n",
+       "\"freeze_backbone\": False\n",
+       "\"freeze_decoder\":  False\n",
+       "\"ignore_index\":    None\n",
+       "\"in_channels\":     3\n",
+       "\"loss\":            ce\n",
+       "\"lr\":              0.001\n",
+       "\"model\":           unet\n",
+       "\"num_classes\":     6\n",
+       "\"num_filters\":     3\n",
+       "\"patience\":        10\n",
+       "\"tmax\":            50"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# validate that the task's hyperparameters are as expected\n",
+    "task.hparams"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Trainer will use only 1 of 8 GPUs because it is running inside an interactive / notebook environment. You may try to set `Trainer(devices=8)` but please note that multi-GPU inside interactive / notebook environments is considered experimental and unstable. Your mileage may vary.\n",
```

Comment:
This is one of the reasons I like to strip the output

Reply:
Not sure what you're saying here but I stripped output anyway!~